R/stats-mlm-tidiers.R

```diff
@@ -0,0 +1,88 @@
+#' @templateVar class lm
+#' @template title_desc_tidy
+#'
+#' @param x An `lm` object created by [stats::lm()].
+#' @template param_confint
+#' @template param_quick
+#' @template param_unused_dots
+#'
+#' @evalRd return_tidy(regression = TRUE)
+#'
+#' @details If the linear model is an `mlm` object (multiple linear model),
+#'   there is an additional column `response`.
+#'
+#'   If you have missing values in your model data, you may need to refit
+#'   the model with `na.action = na.exclude`.
+#'
+#' @examples
+#'
+#' mod <- lm(cbind(mpg, disp) ~ wt, mtcars)
+#' tidy(mod, conf.int = TRUE)
+#' @aliases lm_tidiers
+#' @export
+#' @seealso [tidy()]
+#' @family lm tidiers
+#' @rdname tidy.mlm
+#' @export
+
+tidy.mlm <- function(x,
+                     conf.int = FALSE,
+                     conf.level = .95,
+                     quick = FALSE,
+                     ...) {
+  # if only term and estimate columns are needed
+  if (quick) {
+    co <- stats::coef(x)
+
+    ret <-
+      data.frame(
+        response = rep(colnames(co), each = nrow(co)),
+        term = rep(rownames(co), times = ncol(co)),
+        estimate = as.numeric(co),
+        stringsAsFactors = FALSE
+      )
+
+    return(process_mlm(ret, 
```

Comment:
Similarly, I no longer believe in `process_*` helpers. Tidiers should for the most part be a single function that reads linearly. Can you move the `process_mlm()` functionality into the body of `tidy.mlm()`?

Reply:
The problem is that `process_mlm` function is being used in two different functions and so it will be better to have a separate function to avoid code duplication.

Reply:
When we get rid of the `quick` argument it only gets called in one place.

Reply:
Okay, I will postpone refactoring to a later date then - when this argument is actually retired.